<launch>

<rosparam command="load" file="$(find vi_ekf)/params/gains.yaml" ns="vi_ekf_node"/>
<rosparam command="load" file="$(find vi_ekf)/params/ocam.yaml" ns="vi_ekf_node"/>

<node name="camera" type="ocam_ros_node" pkg="ocam_ros">
  <param name="device_path" value="/dev/video1"/>
  <remap from="image_raw" to="color"/>
</node>

<node name="inertialsense" type="inertial_sense_node" pkg="inertial_sense">
  <param name="stream_IMU" value="true"/>
  <param name="navigation_dt_ms" value="10"/>
</node>

<node name="vi_ekf_node" type="vi_ekf_node" pkg="vi_ekf" output="screen"/>

<node name="vrpn" type="vrpn_client_node" pkg="vrpn_client_ros" output="screen">
  <param name="refresh_tracker_frequency" value="1"/>
  <param name="use_server_time" value="true"/>
  <param name="server" value="192.168.0.103"/>
  <remap from="vrpn/ocam/pose" to="truth/pose"/>
</node>

<node name="tracked_view" type="image_view" pkg="image_view">
	<remap from="image" to="tracked"/>
</node>

<!--<node name="record" type="record" pkg="rosbag" args="-O /home/superjax/rosbag/vi_ekf_results /odom /imu /truth/pose" output="screen"/>-->

<node name="record" type="record" pkg="rosbag" args="-O /home/superjax/$(arg name) /imu /truth/pose /color"/>

</launch>
